### PR TITLE
graph node manager extension is missing

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

* convert to SDK style
* stop referencing packages we don't appear to need.
* fix a crash
* add a todo to improve filter item implementation.
* add logging and put crashing dictionary checks in a try catch.
Screen Shot 2022-06-17 at 10 57 17 AM
Screen Shot 2022-06-17 at 10 57 11 AM

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/GraphNodeManagerViewExtension/GraphNodeManagerViewExtension.csproj [hunk 1]
```diff
@@ -1,79 +1,38 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (pinzart90) [inline src/GraphNodeManagerViewExtension/GraphNodeManagerViewExtension.csproj:1]: hmm...I must have missed this project when first moving to sdk style ?

Comment (mjkkirschner) [reply src/GraphNodeManagerViewExtension/GraphNodeManagerViewExtension.csproj:1]: I think maybe it was not merged yet.

File: src/GraphNodeManagerViewExtension/GraphNodeManagerViewExtension.csproj [hunk 2]
```diff
@@ -83,49 +42,6 @@
     <Reference Include="System.Xml" />
     <Reference Include="WindowsBase" />
   </ItemGroup>
-  <ItemGroup>
-    <Compile Include="Controls\FilterItemControl.xaml.cs">
-      <DependentUpon>FilterItemControl.xaml</DependentUpon>
-    </Compile>
-    <Compile Include="Controls\InfoBubbleControl.xaml.cs">
```
Comment (pinzart90) [inline src/GraphNodeManagerViewExtension/GraphNodeManagerViewExtension.csproj:90]: pls check that the .xaml and .xaml.cs are displayed properly in vs

**Review by QilongTang — APPROVED** [review]

File: src/GraphNodeManagerViewExtension/GraphNodeManagerViewModel.cs [hunk 3]
```diff
@@ -170,9 +171,10 @@ public bool IsAnyFilterOn
         /// Establish the Current Workspace
         /// </summary>
         /// <param name="p"></param>
-        public GraphNodeManagerViewModel(ViewLoadedParams p, string id)
+        public GraphNodeManagerViewModel(ViewLoadedParams p, string id, Action<Logging.ILogMessage> logDelegate)
```
Comment (pinzart90) [inline src/GraphNodeManagerViewExtension/GraphNodeManagerViewModel.cs:174]: is the viewModel ok to break public signature ?

File: src/GraphNodeManagerViewExtension/GraphNodeManagerViewModel.cs [hunk 4]
```diff
@@ -355,61 +357,68 @@ private void NodesCollectionViewSource_Filter(object sender, FilterEventArgs e)
...
-            {
-                e.Accepted = false;
-                return;
-            }
-            if (!nvm.IssuesHasWarning && filterDictionary["Warning"].IsFilterOn)
+            try
             {
-                e.Accepted = false;
-                return;
+                // Boolean Toggle Filters
```
Comment (pinzart90) [inline src/GraphNodeManagerViewExtension/GraphNodeManagerViewModel.cs:362]: wat caused the crash?

File: src/GraphNodeManagerViewExtension/GraphNodeManagerViewModel.cs (hunk 3 above)
Comment (reddyashish) [reply src/GraphNodeManagerViewExtension/GraphNodeManagerViewModel.cs:174]: This is recently added. Not part of previous release.

Comment (pinzart90) [reply src/GraphNodeManagerViewExtension/GraphNodeManagerViewModel.cs:174]: ok

**Review by pinzart90 — APPROVED** [review]

**Comment by mjkkirschner** [comment]
The PR polling job is getting backed up, so kicked this off manually:
https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/

**Comment by QilongTang** [comment]
My applogize.. Did not see the regression
```
<html>
<body>
<!--StartFragment-->

Test Result (1 failure / +1)DynamoCoreWpfTests.PackageManagerExtensionLoadingTests.PackageManagerLoadsAndAddsViewExtension
--


<!--EndFragment-->
</body>
</html>[Test Result](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/6381/testReport/) (1 failure / +1)
[DynamoCoreWpfTests.PackageManagerExtensionLoadingTests.PackageManagerLoadsAndAddsViewExtension](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/6381/testReport/junit/DynamoCoreWpfTests/PackageManagerExtensionLoadingTests/PackageManagerLoadsAndAddsViewExtension/)
```

**Comment by dnenov** [comment]
Hey @mjkkirschner - thank you for all that. I really appreciate it.

**Pull request merged**